clifford/_mvarray.py

```diff
@@ -9,14 +9,38 @@
 call_array = np.vectorize(MultiVector.__call__)
 
 
+def _interrogate_nested_mvs(input_array):
+    """
+    Calculates the shape of the nested input_array, and gets the associated layout.
+    Stops descending when it encounters a MultiVector.
+    """
+    if not isinstance(input_array[0], MultiVector):
+        nested_shape, layout, dtype = _interrogate_nested_mvs(input_array[0])
+        return (len(input_array), *nested_shape), layout, dtype
+    else:
+        return tuple([len(input_array)]), input_array[0].layout, type(input_array[0].value[0])
```

Comment:
Your recursion can have an even simpler base case:
```suggestion
    if not isinstance(input_array, MultiVector):
        nested_shape, layout, dtype = _interrogate_nested_mvs(input_array)
        return (len(input_array), *nested_shape), layout, dtype
    else:
        return (), input_array.layout, input_array.value.dtype
```